daffodil-io/src/main/scala/org/apache/daffodil/io/Dump.scala

```diff
@@ -598,13 +599,13 @@ class DataDumper {
 
     val endByteAddress0b = math.max(startByteAddress0b + lengthInBytes - 1, 0)
     // val cs = optEncodingName.map { Charset.forName(_) }
-    val decoder = getReplacingDecoder(optEncodingName)
+    val decoder = getReportingDecoder(optEncodingName)
     var i = startByteAddress0b
     val sb = new StringBuilder
     while (i <= endByteAddress0b) {
-      val (cR, _, _) = convertToCharRepr(i - startByteAddress0b, endByteAddress0b, byteSource, decoder)
-      sb += cR(0)
-      i += 1
+      val (cR, nBytesConsumed, _) = convertToCharRepr(i - startByteAddress0b, endByteAddress0b, byteSource, decoder)
```

Comment:
I wonder what will happen in this code if we have a text large enough to do debug dumps of, but it is in one of our non-8-bit charset encodings? 

Do we have tests of this, or alternatively, does this code exclude that possibility somewhere? (I imagine the latter is more likely. This kind of dump makes little sense to have text and hex if the characters are not using up units of whole bytes.

Reply:
It looks like we do have code that checks for non byte aligned encodings, and we default to per byte decoding based on windows-1252 in those cases. Depending on where this is called from, we'll either get text and hex (e.g trace output) or text only (e.g left over data dump).

It doesn't look like we have tests for decoding non-8 bit charset encodings.

The update to this code was intended to add multibyte sequence support to the textOnly dump as the original code didn't have support for that.

Reply:
getReplacingDecoder returns a JavaCharsetDecoder, so it can only decode full byte characters. I believe there's a bug to update the text dumper to support non-byte size charsets.

Reply:
Right, we won't actually use the decoder though, due to the code in DataLoc.dump. It looks like it won't return a decoder if it's an unknown encoding or if the known encoding isn't byte aligned.

Reply:
I don't know if we should be trying to enhance this to produce a useful textual dump along side the byte dump for non-8-bit characters. Making that meaningful in any way is hard. I think one would have to do a binary (individual bits) dump, instead of hex to do that. Let's not do that. I think it is overkill.  Or rather, let's not do that as part of fixing THIS jira ticket, which is just about the left-over-data dump.

Reply:
As mentioned above, [DAFFODIL-378](https://issues.apache.org/jira/browse/DAFFODIL-378) was created to address this